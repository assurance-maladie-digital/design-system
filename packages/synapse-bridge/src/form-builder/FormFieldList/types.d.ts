import { Field } from '../FormField/types';

export interface Fields {
	[fieldName: string]: Field;
}

export interface FieldList {
	title?: string;
	description?: string;
	questions: Fields;
}
