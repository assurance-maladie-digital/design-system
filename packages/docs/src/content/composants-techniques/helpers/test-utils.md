---
title: testUtils
description: Le helper `testUtils` contient un ensemble de fonctions utilitaires pour les tests unitaires.
---

<doc-tabs>

<doc-tab-item label="Utilisation">

<doc-alert type="info">

Le helper `testUtils` exporte également toutes les fonctions du package [Vue Test Utils](https://vue-test-utils.vuejs.org/).

</doc-alert>

### applyGlobalConfig

La fonction `applyGlobalConfig` permet d’appliquer la configuration globale pour les tests, elle est utilisée dans le fichier `setup.ts`.

```ts
import { applyGlobalConfig } from '@cnamts/vue-dot/src/helpers/testUtils';

applyGlobalConfig();
```

### installGlobalPlugins

La fonction `installGlobalPlugins` permet d’installer les plugins [Vue Input Facade](https://ronaldjerez.github.io/vue-input-facade/), [Vue Meta](https://vue-meta.nuxtjs.org/), [Vuex](https://vuex.vuejs.org/), [Vuetify](https://v2.vuetifyjs.com/) et Vue Dot pour les tests.

```ts
import { createLocalVue, installGlobalPlugins } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installGlobalPlugins(localVue);
```

### installRouter

La fonction `installRouter` permet d’installer [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createLocalVue, installRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installRouter(localVue);
```

### createRouter

La fonction `createRouter` permet de créer une instance de [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const routes = [];

const router = createRouter({
	routes
});
```

### createStore

La fonction `createStore` permet de créer une instance de [Vuex](https://vuex.vuejs.org/) pour les tests.

```ts
import { createStore } from '@cnamts/vue-dot/src/helpers/testUtils';

const store = createStore({
	state: {}
});
```

### createVuetifyInstance

La fonction `createVuetifyInstance` permet de créer une instance de [Vuetify](https://v2.vuetifyjs.com/) pour les tests.

```ts
import { createVuetifyInstance } from '@cnamts/vue-dot/src/helpers/testUtils';

const vuetify = createVuetifyInstance();
```

### mockVFormRef

La fonction `mockVFormRef` permet de mocker la référence d’un composant [VForm](https://v2.vuetifyjs.com/en/components/forms/) pour les tests.

```ts
import { mockVFormRef } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	$refs: {
		form: mockVFormRef(true)
	}
};
```

### mockTranslations

La fonction `mockTranslations` permet de mocker les traductions pour les tests.

```ts
import { mockTranslations } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	...mockTranslations<string | string[]>({
		'path.to.title': 'Titre',
		'path.to.items': [
			'Élément 1',
			'Élément 2',
			'Élément 3'
		]
	})
};
```

</doc-tab-item>

<doc-tab-item label="API">
<doc-api name="helpers/test-utils"></doc-api>
</doc-tab-item>

</doc-tabs>
