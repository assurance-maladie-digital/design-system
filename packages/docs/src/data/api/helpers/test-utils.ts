import { Api } from '~/types';

export const api: Api = {
	applyGlobalConfig: {
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	installGlobalPlugins: {
		arguments: [
			{
				name: 'localVue',
				description: 'Local Vue instance',
				type: 'Vue',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	installRouter: {
		arguments: [
			{
				name: 'localVue',
				description: 'Local Vue instance',
				type: 'Vue',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	createRouter: {
		arguments: [
			{
				name: 'options',
				description: 'Router options',
				type: 'RouterOptions',
				required: false,
			},
		],
		returnValue: [
			{
				type: 'VueRouter',
			},
		],
	},
	createStore: {
		arguments: [
			{
				name: 'options',
				description: 'Store options',
				type: 'StoreOptions',
				required: false,
			},
		],
		returnValue: [
			{
				type: 'Store',
			},
		],
	},
	createVuetifyInstance: {
		arguments: [
			{
				name: 'preset',
				description: 'Vuetify preset',
				type: 'Partial<UserVuetifyPreset>',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'Vuetify',
			},
		],
	},
	mockVFormRef: {
		arguments: [
			{
				name: 'isValid',
				description: 'Is valid',
				type: 'boolean',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	mockTranslations: {
		arguments: [
			{
				name: 'translations',
				description: 'Translations',
				type: 'IndexedObject',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
};
