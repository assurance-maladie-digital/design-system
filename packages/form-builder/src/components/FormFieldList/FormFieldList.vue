<template>
	<div
		class="vd-field-list w-100"
		:class="{ 'theme--dark': $vuetify.theme.dark }"
	>
		<h3
			v-if="listTitle"
			class="text-h6 font-weight-bold mb-2"
		>
			{{ listTitle }}
		</h3>

		<slot
			name="section-description"
			v-bind="{ description }"
		>
			<p v-if="description">
				{{ description }}
			</p>
		</slot>

		<FormField
			v-for="(field, fieldName) in fields"
			:key="'field-' + fieldName"
			:field="field"
			@change="fieldUpdated($event, fieldName)"
		>
			<!--
				Pass down the scoped slots
				@see https://github.com/vuejs/vue/pull/7765 for the official discussion
				and https://stackoverflow.com/questions/50891858/vue-how-to-pass-down-slots-inside-wrapper-component/52823029#52823029
				for the source of this solution
			-->
			<template
				v-for="slot in Object.keys($scopedSlots)"
				#[slot]="scope"
			>
				<slot
					:name="slot"
					v-bind="scope"
				/>
			</template>
		</FormField>
	</div>
</template>

<script lang="ts">
	import Vue, { PropType } from 'vue';
	import Component, { mixins } from 'vue-class-component';

	import FormField from '../FormField';
	import { Field } from '../FormField/types';

	import { Fields } from './types';

	const Props = Vue.extend({
		props: {
			listTitle: {
				type: String,
				default: undefined
			},
			description: {
				type: String,
				default: undefined
			},
			fields: {
				type: Object as PropType<Fields>,
				default: undefined
			}
		}
	});

	const MixinsDeclaration = mixins(Props);

	@Component({
		components: {
			FormField
		},
		model: {
			prop: 'fields',
			event: 'change'
		}
	})
	export default class FormFieldList extends MixinsDeclaration {
		fieldUpdated(field: Field, fieldName: string): void {
			const fields = { ...this.fields };
			fields[fieldName] = field;

			this.$nextTick(() => {
				this.$emit('change', fields);

				if (field.dynamic) {
					this.$emit('refresh');
				}
			});
		}
	}
</script>
