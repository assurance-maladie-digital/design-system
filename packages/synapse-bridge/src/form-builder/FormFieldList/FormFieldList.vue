<script lang="ts">
import { defineComponent } from 'vue'
import type { PropType } from 'vue'

import FormField from '../FormField'
import type { Field } from '../FormField/types'
import type { Fields } from './types'

import { useTheme } from 'vuetify'

export default defineComponent({
	components: {
		FormField,
	},
	props: {
		listTitle: {
			type: String,
			default: undefined,
		},
		description: {
			type: String,
			default: undefined,
		},
		modelValue: {
			type: Object as PropType<Fields>,
			default: undefined,
		},
	},
	emits: ['update:modelValue', 'refresh'],
	data() {
		return {
			theme: useTheme(),
		}
	},
	methods: {
		fieldUpdated(field: Field, fieldName: string): void {
			const fields = { ...this.modelValue }
			fields[fieldName] = field

			this.$nextTick(() => {
				this.$emit('update:modelValue', fields)

				if (field.dynamic) {
					this.$emit('refresh')
				}
			})
		},
	},
})
</script>

<template>
	<div
		class="vd-field-list w-100"
		:class="{ 'theme--dark': theme.current.dark }"
	>
		<h3 v-if="listTitle" class="text-h6 font-weight-bold mb-2">
			{{ listTitle }}
		</h3>

		<slot name="section-description" v-bind="{ description }">
			<p v-if="description">
				{{ description }}
			</p>
		</slot>

		<FormField
			v-for="(field, fieldName) in modelValue"
			:key="'field-' + fieldName"
			:modelValue="field"
			@update:modelValue="fieldUpdated($event, fieldName as string)"
		>
			<template v-for="slot in Object.keys($slots)" v-slot:[slot]="scope">
				<slot :name="slot" v-bind="scope" />
			</template>
		</FormField>
	</div>
</template>
