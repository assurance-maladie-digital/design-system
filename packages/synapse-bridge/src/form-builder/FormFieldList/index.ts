import FormFieldList from './FormFieldList.vue';

export default FormFieldList;
