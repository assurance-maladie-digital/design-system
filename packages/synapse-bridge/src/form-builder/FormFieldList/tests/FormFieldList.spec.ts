import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import { vuetify } from '@tests/unit/setup'

import FormFieldList from '../'

import { addressFields } from './data/addressFields'

describe('FormFieldList', () => {
	it('renders correctly', () => {
		const wrapper = mount(
			{
				template:
					'<VApp><FormFieldList v-model="addressFields" listTitle="title" /></VApp>',
				data() {
					return {
						addressFields,
						title: 'Address Fields',
					}
				},
				components: {
					FormFieldList,
				},
			},
			{
				global: {
					plugins: [vuetify],
				},
			}
		)
		expect(wrapper).toMatchSnapshot()
	})

	it('emit a update:modelValue event when a field is updated', async () => {
		const wrapper = mount(
			{
				template:
					'<VApp><FormFieldList v-model="addressFields" listTitle="title" /></VApp>',
				data() {
					return {
						addressFields,
					}
				},
				components: {
					FormFieldList,
				},
			},
			{
				global: {
					plugins: [vuetify],
				},
			}
		)

		const input = wrapper.find('input')
		await input.setValue('123')
		const FormFieldListComponent = wrapper.findComponent(FormFieldList)
		expect(FormFieldListComponent.emitted()).toHaveProperty(
			'update:modelValue'
		)
	})

	it('emit a refresh event when a field is updated', async () => {
		const wrapper = mount(
			{
				template:
					'<VApp><FormFieldList v-model="addressFields" /></VApp>',
				data() {
					return {
						addressFields: {
							...addressFields,
							streetNumber: {
								dynamic: true,
								type: 'number',
								value: null,
								fieldOptions: {
									label: 'Num√©ro de voie',
									mask: '#####',
									variant: 'outlined',
								},
							},
						},
					}
				},
				components: {
					FormFieldList,
				},
			},
			{
				global: {
					plugins: [vuetify],
				},
			}
		)

		const input = wrapper.find('input')
		await input.setValue('123')
		const FormFieldListComponent = wrapper.findComponent(FormFieldList)

		expect(FormFieldListComponent.emitted()).toHaveProperty('refresh')
	})
})
