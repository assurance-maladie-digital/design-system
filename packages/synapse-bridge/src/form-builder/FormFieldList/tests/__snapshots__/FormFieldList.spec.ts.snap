// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormFieldList > renders correctly 1`] = `
<div class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr">
  <div class="v-application__wrap">
    <div class="vd-field-list w-100 pb-8">
      <h3 class="text-h6 font-weight-bold mb-2">title</h3>
      <!--v-if-->
      <div data-v-9067cb71="" class="vd-form-field">
        <!--v-if-->
        <!--v-if-->
        <div data-v-9067cb71="" class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field vd-form-input">
          <!---->
          <div class="v-input__control">
            <div class="v-field v-field--center-affix v-field--variant-outlined v-theme--light v-locale--is-ltr">
              <div class="v-field__overlay"></div>
              <div class="v-field__loader">
                <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                  <!---->
                  <div class="v-progress-linear__background bg-primary" style="width: 100%;"></div>
                  <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                    <div class="v-progress-linear__indeterminate">
                      <div class="v-progress-linear__indeterminate long bg-primary"></div>
                      <div class="v-progress-linear__indeterminate short bg-primary"></div>
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div class="v-field__field" data-no-activator="">
                <!----><label class="v-label v-field-label" for="input-0">
                  <!---->Numéro de voie
                </label>
                <!----><input size="1" type="text" id="input-0" aria-describedby="input-0-messages" mask="#####" class="v-field__input">
                <!---->
              </div>
              <!---->
              <!---->
              <div class="v-field__outline">
                <div class="v-field__outline__start"></div>
                <div class="v-field__outline__notch"><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-0">
                    <!---->Numéro de voie
                  </label></div>
                <div class="v-field__outline__end"></div>
                <!---->
              </div>
            </div>
          </div>
          <!---->
          <div class="v-input__details">
            <transition-stub name="slide-y-transition" appear="false" persisted="false" css="true" tag="div" class="v-messages" role="alert" aria-live="polite" id="input-0-messages">
              <!---->
            </transition-stub>
            <!---->
          </div>
        </div>
      </div>
      <div data-v-9067cb71="" class="vd-form-field">
        <!--v-if-->
        <!--v-if-->
        <div data-v-60156eea="" data-v-9067cb71="">
          <div data-v-60156eea="" class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field v-select v-select--single vd-form-input">
            <!---->
            <div class="v-input__control">
              <div class="v-field v-field--appended v-field--center-affix v-field--variant-outlined v-theme--light v-locale--is-ltr" role="combobox">
                <div class="v-field__overlay"></div>
                <div class="v-field__loader">
                  <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                    <!---->
                    <div class="v-progress-linear__background bg-primary" style="width: 100%;"></div>
                    <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                      <div class="v-progress-linear__indeterminate">
                        <div class="v-progress-linear__indeterminate long bg-primary"></div>
                        <div class="v-progress-linear__indeterminate short bg-primary"></div>
                      </div>
                    </transition-stub>
                    <!---->
                  </div>
                </div>
                <!---->
                <div class="v-field__field" data-no-activator="">
                  <!----><label class="v-label v-field-label" for="input-2">
                    <!---->Complément de voie
                  </label>
                  <!---->
                  <div class="v-field__input" data-no-activator="">
                    <!---->
                    <!----><input size="1" type="text" id="input-2" aria-describedby="input-2-messages" inputmode="none" aria-label="Open" title="Open" value="">
                  </div>
                  <!---->
                </div>
                <!---->
                <div class="v-field__append-inner">
                  <!----><i class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-select__menu-icon" aria-hidden="true"></i>
                  <!---->
                </div>
                <div class="v-field__outline">
                  <div class="v-field__outline__start"></div>
                  <div class="v-field__outline__notch"><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-2">
                      <!---->Complément de voie
                    </label></div>
                  <div class="v-field__outline__end"></div>
                  <!---->
                </div>
              </div>
            </div>
            <!---->
            <div class="v-input__details">
              <transition-stub name="slide-y-transition" appear="false" persisted="false" css="true" tag="div" class="v-messages" role="alert" aria-live="polite" id="input-2-messages">
                <!---->
              </transition-stub>
              <!---->
            </div>
          </div>
          <!--v-if-->
        </div>
      </div>
      <div data-v-9067cb71="" class="vd-form-field">
        <!--v-if-->
        <!--v-if-->
        <div data-v-60156eea="" data-v-9067cb71="">
          <div data-v-60156eea="" class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field v-autocomplete v-autocomplete--single vd-form-input">
            <!---->
            <div class="v-input__control">
              <div class="v-field v-field--appended v-field--center-affix v-field--variant-outlined v-theme--light v-locale--is-ltr" role="combobox">
                <div class="v-field__overlay"></div>
                <div class="v-field__loader">
                  <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                    <!---->
                    <div class="v-progress-linear__background bg-primary" style="width: 100%;"></div>
                    <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                      <div class="v-progress-linear__indeterminate">
                        <div class="v-progress-linear__indeterminate long bg-primary"></div>
                        <div class="v-progress-linear__indeterminate short bg-primary"></div>
                      </div>
                    </transition-stub>
                    <!---->
                  </div>
                </div>
                <!---->
                <div class="v-field__field" data-no-activator="">
                  <!----><label class="v-label v-field-label" for="input-5">
                    <!---->Type de voie
                  </label>
                  <!---->
                  <div class="v-field__input" data-no-activator="">
                    <!---->
                    <!----><input size="1" type="text" id="input-5" aria-describedby="input-5-messages" value="">
                  </div>
                  <!---->
                </div>
                <!---->
                <div class="v-field__append-inner">
                  <!----><i class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable v-autocomplete__menu-icon" role="button" aria-hidden="false" tabindex="0" aria-label="Open" title="Open"></i>
                  <!---->
                </div>
                <div class="v-field__outline">
                  <div class="v-field__outline__start"></div>
                  <div class="v-field__outline__notch"><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-5">
                      <!---->Type de voie
                    </label></div>
                  <div class="v-field__outline__end"></div>
                  <!---->
                </div>
              </div>
            </div>
            <!---->
            <div class="v-input__details">
              <transition-stub name="slide-y-transition" appear="false" persisted="false" css="true" tag="div" class="v-messages" role="alert" aria-live="polite" id="input-5-messages">
                <!---->
              </transition-stub>
              <!---->
            </div>
          </div>
          <!--v-if-->
        </div>
      </div>
      <div data-v-9067cb71="" class="vd-form-field">
        <!--v-if-->
        <!--v-if-->
        <div data-v-9067cb71="" class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field vd-form-input">
          <!---->
          <div class="v-input__control">
            <div class="v-field v-field--center-affix v-field--variant-outlined v-theme--light v-locale--is-ltr">
              <div class="v-field__overlay"></div>
              <div class="v-field__loader">
                <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                  <!---->
                  <div class="v-progress-linear__background bg-primary" style="width: 100%;"></div>
                  <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                    <div class="v-progress-linear__indeterminate">
                      <div class="v-progress-linear__indeterminate long bg-primary"></div>
                      <div class="v-progress-linear__indeterminate short bg-primary"></div>
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div class="v-field__field" data-no-activator="">
                <!----><label class="v-label v-field-label" for="input-8">
                  <!---->Libellé de la voie
                </label>
                <!----><input size="1" type="text" id="input-8" aria-describedby="input-8-messages" class="v-field__input">
                <!---->
              </div>
              <!---->
              <!---->
              <div class="v-field__outline">
                <div class="v-field__outline__start"></div>
                <div class="v-field__outline__notch"><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-8">
                    <!---->Libellé de la voie
                  </label></div>
                <div class="v-field__outline__end"></div>
                <!---->
              </div>
            </div>
          </div>
          <!---->
          <div class="v-input__details">
            <transition-stub name="slide-y-transition" appear="false" persisted="false" css="true" tag="div" class="v-messages" role="alert" aria-live="polite" id="input-8-messages">
              <!---->
            </transition-stub>
            <!---->
          </div>
        </div>
      </div>
      <div data-v-9067cb71="" class="vd-form-field">
        <!--v-if-->
        <!--v-if-->
        <div data-v-9067cb71="" class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field vd-form-input">
          <!---->
          <div class="v-input__control">
            <div class="v-field v-field--center-affix v-field--variant-outlined v-theme--light v-locale--is-ltr">
              <div class="v-field__overlay"></div>
              <div class="v-field__loader">
                <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                  <!---->
                  <div class="v-progress-linear__background bg-primary" style="width: 100%;"></div>
                  <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                    <div class="v-progress-linear__indeterminate">
                      <div class="v-progress-linear__indeterminate long bg-primary"></div>
                      <div class="v-progress-linear__indeterminate short bg-primary"></div>
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div class="v-field__field" data-no-activator="">
                <!----><label class="v-label v-field-label" for="input-10">
                  <!---->Complément d’adresse
                </label>
                <!----><input size="1" type="text" id="input-10" aria-describedby="input-10-messages" class="v-field__input">
                <!---->
              </div>
              <!---->
              <!---->
              <div class="v-field__outline">
                <div class="v-field__outline__start"></div>
                <div class="v-field__outline__notch"><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-10">
                    <!---->Complément d’adresse
                  </label></div>
                <div class="v-field__outline__end"></div>
                <!---->
              </div>
            </div>
          </div>
          <!---->
          <div class="v-input__details">
            <transition-stub name="slide-y-transition" appear="false" persisted="false" css="true" tag="div" class="v-messages" role="alert" aria-live="polite" id="input-10-messages">
              <!---->
            </transition-stub>
            <!---->
          </div>
        </div>
      </div>
      <div data-v-9067cb71="" class="vd-form-field">
        <!--v-if-->
        <!--v-if-->
        <div data-v-9067cb71="" class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field vd-form-input">
          <!---->
          <div class="v-input__control">
            <div class="v-field v-field--center-affix v-field--variant-outlined v-theme--light v-locale--is-ltr">
              <div class="v-field__overlay"></div>
              <div class="v-field__loader">
                <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                  <!---->
                  <div class="v-progress-linear__background bg-primary" style="width: 100%;"></div>
                  <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                    <div class="v-progress-linear__indeterminate">
                      <div class="v-progress-linear__indeterminate long bg-primary"></div>
                      <div class="v-progress-linear__indeterminate short bg-primary"></div>
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div class="v-field__field" data-no-activator="">
                <!----><label class="v-label v-field-label" for="input-12">
                  <!---->Code postal
                </label>
                <!----><input size="1" type="text" id="input-12" aria-describedby="input-12-messages" mask="#####" class="v-field__input">
                <!---->
              </div>
              <!---->
              <!---->
              <div class="v-field__outline">
                <div class="v-field__outline__start"></div>
                <div class="v-field__outline__notch"><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-12">
                    <!---->Code postal
                  </label></div>
                <div class="v-field__outline__end"></div>
                <!---->
              </div>
            </div>
          </div>
          <!---->
          <div class="v-input__details">
            <transition-stub name="slide-y-transition" appear="false" persisted="false" css="true" tag="div" class="v-messages" role="alert" aria-live="polite" id="input-12-messages">
              <!---->
            </transition-stub>
            <!---->
          </div>
        </div>
      </div>
      <div data-v-9067cb71="" class="vd-form-field">
        <!--v-if-->
        <!--v-if-->
        <div data-v-9067cb71="" class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field vd-form-input">
          <!---->
          <div class="v-input__control">
            <div class="v-field v-field--center-affix v-field--variant-outlined v-theme--light v-locale--is-ltr">
              <div class="v-field__overlay"></div>
              <div class="v-field__loader">
                <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                  <!---->
                  <div class="v-progress-linear__background bg-primary" style="width: 100%;"></div>
                  <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                    <div class="v-progress-linear__indeterminate">
                      <div class="v-progress-linear__indeterminate long bg-primary"></div>
                      <div class="v-progress-linear__indeterminate short bg-primary"></div>
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div class="v-field__field" data-no-activator="">
                <!----><label class="v-label v-field-label" for="input-14">
                  <!---->Ville
                </label>
                <!----><input size="1" type="text" id="input-14" aria-describedby="input-14-messages" class="v-field__input">
                <!---->
              </div>
              <!---->
              <!---->
              <div class="v-field__outline">
                <div class="v-field__outline__start"></div>
                <div class="v-field__outline__notch"><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-14">
                    <!---->Ville
                  </label></div>
                <div class="v-field__outline__end"></div>
                <!---->
              </div>
            </div>
          </div>
          <!---->
          <div class="v-input__details">
            <transition-stub name="slide-y-transition" appear="false" persisted="false" css="true" tag="div" class="v-messages" role="alert" aria-live="polite" id="input-14-messages">
              <!---->
            </transition-stub>
            <!---->
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
